### ISSUE TITLE
"Unexpected token :" on unassigned object

### ISSUE THREAD

**Issue by emilbayes** [eslint/eslint#1994]
I get the error `Unexpected token :` on the following code:

```js
{op: 'hello', path: '/a/b/c'};
```

I realise that it is not valid Javascript to have unassigned (?) objects floating around like that, but maybe a rule could be made, to make a more meaningful error message. You're welcome to mark as `wontfix` if this is too much of a rarity to run into this kind of problem.

**Comment by lo1tuma** [comment]
This is a syntax error reported from the parser.
Whenever a parser error occurs, eslint will not execute any rule for this
code. So it is not possible to add a rule which detects such patterns and
report a different error message.

If you want better error messages from the parser I suggest that you file
an issue at eslint/espree.

**Comment by nzakas** [comment]
The parser follows V8 error message formats, you'd get the same error in the browser.
